vernac/classes.ml

```diff
@@ -152,6 +152,28 @@ let default_locality () =
 let instance_locality =
   Attributes.hint_locality ~default:default_locality
 
+type event =
+  | NewClass of typeclass
+  | NewInstance of {
+      class_name : GlobRef.t;
+      instance : GlobRef.t;
+      info : Typeclasses.hint_info;
+      locality : Hints.hint_locality;
+    }
+
+let observers = ref CString.Map.empty
+
+let add_observer ~name o = observers := CString.Map.add name o !observers
```

Comment:
What is the expected behavior if a name is reused? Currently the last to add_observer wins, is that what we want?

Reply:
Good point, I'll fix that tomorrow